src/fastoad/notebooks/02_explanation/04_OpenMDAO.ipynb

```diff
@@ -0,0 +1,892 @@
+{
```

Comment:
<blockquote>For each input/output we want to add to our component we must know</blockquote><p>--&gt; "...to our component<code><strong>,</strong></code> we must know" (of course, the code block format is only to highlight the modification)</p><p> </p><blockquote>... since the name will be reused in some other components</blockquote><p>Only because we chose the strategy to promote all variables. I think we should mention somewhere that there are <a href="https://openmdao.org/newdocs/versions/latest/basic_user_guide/multidisciplinary_optimization/linking_vars.html" target="_blank">two ways</a> and that we purposely talk only about promotion.</p><p> </p><blockquote>a more detailed explanation of the variable names is available&nbsp;<a href="https://fast-oad.readthedocs.io/en/v1.3.1/documentation/custom_modules/add_modules.html?highlight=naming#variable-naming" target="_blank">here</a></blockquote><p>Please link to https://fast-oad.readthedocs.io/en/stable/documentation/custom_modules/add_modules.html#variable-naming</p><p> </p><blockquote>&nbsp;Different names will be seen as different value so some changes on the aircraft parameters might not be fully repercuted</blockquote><p>&nbsp;--&gt; "Different names will be seen as different value<code>,</code> so some changes ..." (it may look futile, but commas really help the reading)</p><p> </p><blockquote>even if, in reality they might mean the same thing.</blockquote><p>--&gt; "even if, in reality<code>,</code> they might mean the same thing."</p><p>Also, I wonder if it is needed to explain so much the promotion mechanism. Maybe it could be summarized.</p><p> </p><blockquote>OpenMDAO supports automatic unit conversion. Depending on the model we want to implement, the units might be in the imperial or metric system while for a component further down in the computation it is the opposite. Of course, the conversion can be done inside the component but OpenMDAO handles it automatically as long as the units are compatible.</blockquote><p>I suggest "OpenMDAO supports automatic unit conversion. Therefore, different components that work with a same variable may use its value in different units without having to explicitly do the conversion."</p><p> </p><blockquote>The shape of the variable : OpenMDAO can handle variables under the form of an array but the size of said array must be given. Alternatively, the shape_by_conn option can allow us to not fill in that field and let OpenMDAO handle the size on its own based on connected variables and copied shapes (functionnality for more advanced users).</blockquote><p>I suggest "The shape of the variable : by default, OpenMDAO variables are (1,)-shaped numpy arrays. They can have a different shape, which shall be declared in&nbsp;<code>setup</code>. Yet an&nbsp;<a href="https://openmdao.org/newdocs/versions/latest/features/experimental/dyn_shapes.html?highlight=shape_by_conn" target="_blank">advanced feature</a>&nbsp;allows a variable shape to be defined at runtime, based on a shape defined in another component."</p><p> </p><blockquote>The default value of the variable : If the variable hasn't been computed beforehand, OpenMDAO will give it its value by default. Usually, if the problem is properly set up there is no need to give it something else than a NaN, except for variables which we will loop on to give them an initial estimate (for instance the MTOW in our problem, the reason why is further explained in the following remarks).</blockquote><p>I suggest "This is the value OpenMDAO will use if the variable hasn't been computed beforehand. I may be a quick way of providing input values, but there are more versatile solutions, so we generally use <code>numpy.nan</code>" (you even show later that loop variables do not need a real default value)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fast-aircraft-design/FAST-OAD/pull/426/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Reply:
Changed in f33eb885